Source/DafnyCore/Dafny.atg

```diff
@@ -3820,44 +3820,49 @@ ImpliesExpression<out Expression e0, bool allowLemma, bool allowLambda, bool all
 /*------------------------------------------------------------------------*/
 LogicalExpression<out Expression e0, bool allowLemma, bool allowLambda, bool allowBitwiseOps>
 = (. Contract.Ensures(Contract.ValueAtReturn(out e0) != null); IToken/*!*/ x;  Expression/*!*/ e1;
+     Expression first; IToken prefixOp;
      e0 = dummyExpr; /* mute the warning */
   .)
-  ( AndOp                                                       (. x = t; .)
-    RelationalExpression<out e0, allowLemma, allowLambda, allowBitwiseOps>
-    // We conjoin "true" to the left, for two reasons. One reason is to keep track of the source
-    // location of the initial AndOp operator. The other reason is to make, when there is only
-    // one conjunct, that the type checker still checks it to be a boolean.
-    (. e0 = new BinaryExpr(x, BinaryExpr.Opcode.And, new LiteralExpr(new AutoGeneratedToken(x), true), e0); .)
+  ( AndOp                                                       (. x = t; prefixOp = t; .)
+    RelationalExpression<out e0, allowLemma, allowLambda, allowBitwiseOps> (. first = e0; .)
     { IF(IsAndOp())  /* read a conjunction as far as possible */
       AndOp                                                     (. x = t; .)
-      RelationalExpression<out e1, allowLemma, allowLambda, allowBitwiseOps>      (. e0 = new BinaryExpr(x, BinaryExpr.Opcode.And, e0, e1); .)
+      RelationalExpression<out e1, allowLemma, allowLambda, allowBitwiseOps>      (. e0 = new BinaryExpr(x, BinaryExpr.Opcode.And, e0, e1, prefixOp); .)
```

Comment:
Yes right, it's the outermost && that gets the prefix, as you guessed &&1 ((a &&2 b) &&1 c)